src/consumer.rs

```diff
@@ -137,6 +141,130 @@ impl PartitionConsumerJS {
 
         Ok(())
     }
+
+    #[node_bindgen(mt)]
+    async fn get_iterator(
+        &self,
+        offset: OffsetWrapper,
+    ) -> Result<PartitionConsumerIterator, FluvioError> {
+        if let Some(client) = &self.inner {
+            let stream = client.stream(offset.0).await?;
+            let mut foo = PartitionConsumerIterator::new();
+            foo.set_inner(Box::pin(stream));
+            Ok(foo)
+        } else {
+            Err(FluvioError::Other(CLIENT_NOT_FOUND_ERROR_MSG.to_owned()))
+        }
+    }
+}
+
+type PartitionConsumerIteratorInner = Pin<Box<dyn Stream<Item = Result<Record, FluvioError>>>>;
+
+pub struct PartitionConsumerIterator {
+    inner: Option<PartitionConsumerIteratorInner>,
+}
+
+#[node_bindgen]
+impl PartitionConsumerIterator {
+    #[node_bindgen(constructor)]
+    pub fn new() -> Self {
+        Self { inner: None }
+    }
+    pub fn set_inner(&mut self, client: PartitionConsumerIteratorInner) {
+        self.inner.replace(client);
+    }
+
+    #[node_bindgen]
+    fn next(&mut self) -> NextIter {
+        fluvio_future::task::run_block_on(async {
```

Comment:
I think this should return future or something like that.

Reply:
Yeah, https://github.com/infinyon/node-bindgen/issues/132 seems to be the issue.